services/device-registry-mongodb-quarkus/src/main/java/org/eclipse/hono/deviceregistry/mongodb/quarkus/AmqpServerFactory.java

```diff
@@ -0,0 +1,273 @@
+import org.eclipse.hono.deviceregistry.mongodb.config.MongoDbBasedCredentialsConfigProperties;
+import org.eclipse.hono.deviceregistry.mongodb.config.MongoDbBasedTenantsConfigProperties;
+import org.eclipse.hono.deviceregistry.mongodb.model.CredentialsDao;
+import org.eclipse.hono.deviceregistry.mongodb.model.DeviceDao;
+import org.eclipse.hono.deviceregistry.mongodb.model.TenantDao;
+import org.eclipse.hono.deviceregistry.mongodb.service.MongoDbBasedCredentialsService;
+import org.eclipse.hono.deviceregistry.mongodb.service.MongoDbBasedRegistrationService;
+import org.eclipse.hono.deviceregistry.mongodb.service.MongoDbBasedTenantService;
+import org.eclipse.hono.deviceregistry.server.DeviceRegistryAmqpServer;
+import org.eclipse.hono.deviceregistry.service.device.AutoProvisionerConfigProperties;
+import org.eclipse.hono.deviceregistry.service.device.EdgeDeviceAutoProvisioner;
+import org.eclipse.hono.deviceregistry.service.tenant.TenantInformationService;
+import org.eclipse.hono.deviceregistry.util.ServiceClientAdapter;
+import org.eclipse.hono.service.HealthCheckServer;
+import org.eclipse.hono.service.amqp.AmqpEndpoint;
+import org.eclipse.hono.service.credentials.CredentialsService;
+import org.eclipse.hono.service.credentials.DelegatingCredentialsAmqpEndpoint;
+import org.eclipse.hono.service.management.credentials.CredentialsManagementService;
+import org.eclipse.hono.service.management.device.DeviceAndGatewayAutoProvisioner;
+import org.eclipse.hono.service.management.device.DeviceManagementService;
+import org.eclipse.hono.service.registration.DelegatingRegistrationAmqpEndpoint;
+import org.eclipse.hono.service.registration.RegistrationService;
+import org.eclipse.hono.service.tenant.DelegatingTenantAmqpEndpoint;
+import org.eclipse.hono.service.tenant.TenantService;
+
+import io.opentracing.Tracer;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.ConfigMapping.NamingStrategy;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.proton.sasl.ProtonSaslAuthenticatorFactory;
+
+/**
+ * A factory for creating AMQP 1.0 based endpoints of Hono's south bound APIs.
+ *
+ */
+@ApplicationScoped
+public class AmqpServerFactory {
+
+    @Inject
+    Vertx vertx;
+
+    @Inject
+    Tracer tracer;
+
+    @Inject
+    DeviceDao deviceDao;
+
+    @Inject
+    CredentialsDao credentialsDao;
+
+    @Inject
+    DeviceManagementService deviceManagementService;
+
+    @Inject
+    MongoDbBasedCredentialsConfigProperties credentialsServiceProperties;
+
+    @Inject
+    CredentialsManagementService credentialsManagementService;
+
+    @Inject
+    TenantInformationService tenantInformationService;
+
+    @Inject
+    HealthCheckServer healthCheckServer;
+
+    @Inject
+    ProtonSaslAuthenticatorFactory saslAuthenticatorFactory;
+
+    @Inject
+    MessagingKafkaProducerConfigProperties eventKafkaProducerConfig;
+
+    @Inject
+    NotificationKafkaProducerConfigProperties notificationKafkaProducerConfig;
```

Comment:
FMPOV `notificationKafkaProducerConfig` is  not needed here as the notification is being taken care in `NotificationSenderProducer`.